src/v/kafka/requests/fetch_request.h

```diff
@@ -336,21 +336,60 @@ std::ostream& operator<<(std::ostream&, const fetch_response&);
  * Fetch operation context
  */
 struct op_context {
+    class response_iterator {
+    public:
+        using difference_type = void;
+        using pointer = fetch_response::iterator::pointer;
+        using reference = fetch_response::iterator::reference;
+        using iterator_category = std::forward_iterator_tag;
+
+        response_iterator(fetch_response::iterator, op_context* ctx);
+
+        reference operator*() noexcept { return *_it; }
+
+        pointer operator->() noexcept { return &(*_it); }
+
+        response_iterator& operator++();
+
+        const response_iterator operator++(int);
+
+        bool operator==(const response_iterator& o) const noexcept;
+
+        bool operator!=(const response_iterator& o) const noexcept;
```

Comment:
i don't see these implemented. only defined.

Reply:
Those methods are defined in `fetch_request.cc`